third_party/libtock-drivers/src/crp.rs

```diff
@@ -0,0 +1,52 @@
+use crate::result::TockResult;
+use libtock_core::syscalls;
+
+const DRIVER_NUMBER: usize = 0x00008;
+
+mod command_nr {
+    pub const AVAILABLE: usize = 0;
+    pub const GET_PROTECTION: usize = 1;
+    pub const SET_PROTECTION: usize = 2;
+}
+
+#[derive(PartialOrd, PartialEq)]
+pub enum ProtectionLevel {
+    /// Unsupported feature
+    Unknown = 0,
+    /// This should be the factory default for the chip.
+    NoProtection = 1,
+    /// At this level, only JTAG/SWD are disabled but other debugging
+    /// features may still be enabled.
+    JtagDisabled = 2,
+    /// This is the maximum level of protection the chip supports.
+    /// At this level, JTAG and all other features are expected to be
+    /// disabled and only a full chip erase may allow to recover from
+    /// that state.
+    FullyLocked = 0xff,
+}
+
+impl From<usize> for ProtectionLevel {
+    fn from(value: usize) -> Self {
+        match value {
+            1 => ProtectionLevel::NoProtection,
+            2 => ProtectionLevel::JtagDisabled,
+            0xff => ProtectionLevel::FullyLocked,
+            _ => ProtectionLevel::Unknown,
+        }
+    }
+}
```

Comment:
Probably not actionable, but ideally this kind of stuff should be in a shared library between the kernel and user-land. I don't know if Tock already has this kind of shared library.

Reply:
Ah ok, sounds good then.

Reply:
I don't recall such shared library, but agree that it could be useful. CC @jrvanwhy

Reply:
I do not believe there is any such library. If I recall my past communications with other Tock core WG members correctly, they had an explanation for why they don't want to share code between the kernel and `libtock-rs`.

How the OpenSK project structures code in its own repository is up to OpenSK, but that may not be helpful if you would like to upstream this driver in the future.

Discussing this on the Tock core WG call is an option as well.

Reply:
> they don't want to share code

Just to clarify, the idea here is not to share code but to share APIs (e.g. types, enum values), i.e. share what is used to define the interface of kernel and user-land.